sqld/src/hrana/session.rs

```diff
@@ -46,13 +54,27 @@ pub enum ResponseError {
     },
 }
 
-pub async fn handle_hello(_jwt: Option<String>) -> Result<Session> {
-    // TODO: handle the jwt
+pub(super) async fn handle_hello(server: &Server, jwt: Option<String>) -> Result<Session> {
+    if let Some(jwt_key) = server.jwt_key.as_ref() {
+        let Some(jwt) = jwt else {
+            bail!(ResponseError::AuthJwtRequired)
+        };
+        validate_jwt(jwt_key, &jwt)?;
+    }
+
     Ok(Session {
         streams: HashMap::new(),
     })
 }
 
+fn validate_jwt(jwt_key: &jsonwebtoken::DecodingKey, jwt: &str) -> Result<()> {
+    let validation = jsonwebtoken::Validation::new(jsonwebtoken::Algorithm::EdDSA);
+    match jsonwebtoken::decode::<serde_json::Value>(jwt, jwt_key, &validation) {
+        Ok(_token) => Ok(()),
+        Err(error) => bail!(ResponseError::AuthJwtRejected { source: dbg!(error) }),
```

Comment:
```suggestion
        Err(error) => bail!(ResponseError::AuthJwtRejected { source: error }),
```

Reply:
Thanks, _of course_ I forgot to remove this :D